tensorboard/components/tensor_widget/numeric-helper-test.ts

```diff
@@ -0,0 +1,164 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {numericValueToString} from './numeric-helper';
+
+describe('numericValueToString', () => {
+  it('NaN', () => {
+    expect(numericValueToString(NaN)).toEqual('NaN');
+    const decimalPlaces = 4;
+    expect(numericValueToString(NaN, decimalPlaces)).toEqual('NaN');
+    expect(numericValueToString(NaN, decimalPlaces, 'exponential'))
+        .toEqual('NaN');
+  });
+
+  it('Infinity', () => {
+    expect(numericValueToString(-Infinity)).toEqual('-∞');
+    expect(numericValueToString(Infinity)).toEqual('+∞');
+    const decimalPlaces = 4;
+    expect(numericValueToString(-Infinity, decimalPlaces)).toEqual('-∞');
+    expect(numericValueToString(Infinity, decimalPlaces)).toEqual('+∞');
+    expect(numericValueToString(-Infinity, decimalPlaces, 'exponential'))
+        .toEqual('-∞');
+    expect(numericValueToString(Infinity, decimalPlaces, 'exponential'))
+        .toEqual('+∞');
+  });
+
+  it('zero', () => {
+    const x = 0;
+    expect(numericValueToString(x)).toEqual('0.00');
+
+    let decimalPlaces = 0;
+    expect(numericValueToString(x, decimalPlaces)).toEqual('0');
+
+    decimalPlaces = 1;
+    expect(numericValueToString(x, decimalPlaces)).toEqual('0.0');
+
+    decimalPlaces = 0;
+    expect(numericValueToString(x, decimalPlaces, 'exponential'))
+        .toEqual('0e+0');
+    decimalPlaces = 2;
+    expect(numericValueToString(x, decimalPlaces, 'exponential'))
+        .toEqual('0.00e+0');
+  });
+
+  it('Large positive value', () => {
```

Comment:
Comments about naming test in general: mocha, especially when using the `it` syntax, is supposed to be BDD-style. Please describe expected behavior, not just short description like `zero` or `large negative value` (also consistency in capitalization would be nice).